test/support/socket_stub.ex

```diff
@@ -63,56 +52,46 @@ defmodule Mobius.Stubs.Socket do
       url: Keyword.fetch!(opts, :url),
       query: Keyword.fetch!(opts, :query),
       shard: Keyword.fetch!(opts, :shard),
-      test_pid: nil,
-      messages: []
+      test_pid: nil
     }
 
     {:ok, state}
   end
 
   @impl GenServer
   def handle_cast({:send, payload}, state) do
-    {:noreply, update_in(state.messages, &(&1 ++ [{:msg, payload}]))}
+    unless state.test_pid == nil do
```

Comment:
```suggestion
    if state.test_pid != nil do
```

This is really personal, but I'm not a big fan of `unless`. I guess it would be alright for cases where you have to do `if not some_boolean_function()` but here I have to stop for a second to make sure I get what's going on.